content/applications/finance/fiscal_localizations/kenya.rst

```diff
@@ -7,114 +7,320 @@ Kenya
+check the missing elements by clicking :guilabel:`View Product(s)`.
+
+.. image:: kenya/product-registration.png
+   :alt: Product registration template.
+
+Stock movements
+===============
+
+All **stock movements** must be sent to the KRA. They do not require an invoice if they are
+**internal operations** or **stock adjustments**. Therefore, Odoo automatically sends them if the
+following conditions are met:
+
+#. No partners are set for the move; or
+#. The partner is your main company or a branch of the main company.
+
+If the stock moves are **external** (e.g., to someone external the main company or its branches),
+the stock moves are automatically sent *after* the invoice is sent to eTIMS.
+
+.. note::
+   - The stock move must be confirmed before sending the invoice to eTIMS;
+   - The product(s) must be registered for the stock move to be sent to eTIMS. If not, a yellow
+     banner appears to prompt registration.
+
+Purchases
+=========
+
+Odoo automatically fetches new vendor bills from eTIMS every day. Users need to confirm the fetched
+vendor bills and the confirmation to the KRA. Every bill confirmation must be associated with a
+validated receipt. To do so, each bill must be linked to purchase order lines.
+
+.. _kenya/purchases:
+
+In the case of purchases **(not customs imports)**, the steps to link purchase order lines with
+bills are the following:
+
+#. The vendor bill is fetched from the KRA servers. The JSON file is available in the chatter of the
+   vendor bill if needed;
+#. Odoo looks at the **PIN** (tax ID) of the partner;
+
+   - If it is unknown, a new partner is created;
+   - If it is known and the branch ID is the same, Odoo uses the known partner.
+
+#. The user must match the fetched vendor bill with an existing product. Each vendor bill must
+   contain a product to be confirmed and sent to eTIMS later on.
+#. Odoo looks at the existing purchase order lines matching the product and partner (if any). The
+   user must select the correct related purchase order line from the
+   :guilabel:`Purchase Order Line`: dropdown list. The received quantities on the purchase order
+   must be on the same as those indicated on the vendor bill. If not, the vendor bill cannot be sent
+   to eTIMS (indicated by a yellow banner).
+
+   - If no existing purchase order lines match the fetched invoice lines, users can create a
+     purchase order based on the unmatched lines. The resulting stock move *must* be validated. Once
+     done, users can confirm and send the bill to eTIMS.
+
+#. Make sure to set a method in the :guilabel:`eTIMS Payment Method` field of the bill to send the
+   vendor bill to the KRA.
+#. Once all steps are completed, the vendor bill can be sent by clicking :guilabel:`Send to eTIMS`.
+   When the vendor bill has been confirmed on eTIMS, the **KRA invoice number** can be found under
+   the :guilabel:`eTIMS Details`.
+
+.. image:: kenya/purchase-order-lines.png
+   :alt: Bill registration steps.
```

Comment:
The goal is to show the screen of the whole process. It's hard to crop it further than what I did here, because I want to show both the Product field (for product matching) and the Purchase Order Line field (for purchase order matching). I also want to show the banner to reflect what the user would see in this case. Let me know what you think about this screenshot